java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt

```diff
@@ -1165,65 +1257,70 @@ OLD: KE1
             callable,
             enclosingStmt
         )
+*/
+context(KaSession)
+private fun KotlinFileExtractor.extractLoop(
+    loop: KtLoopExpression,
+    bodyIdx: Int?,
+    stmtExprParent: StmtExprParent,
+    callable: Label<out DbCallable>,
+    getId: (Label<out DbStmtparent>, Int) -> Label<out DbStmt>
+): Label<out DbStmt> {
+    val stmtParent = stmtExprParent.stmt(loop, callable)
+    val locId = tw.getLocation(loop)
+
+    val idx: Int
+    val parent: Label<out DbStmtparent>
+
+    val label = (loop.parent as? KtLabeledExpression)?.getLabelName()
+    if (label != null) {
```

Comment:
This follows the logic of the old code, but with the PSI representation it would be more natural to me to write this as
```
val loopParent = loop.parent
if (loopParent is KtLabeledExpression) {
    val label = loopParent.getLabelName()
```

Reply:
I think we'll remove this altogether later, and handle `KtLabeledExpression` explicitly in `extractExpression`.

Reply:
Might be worth adding a TODO comment, so we don't forget about it, then?

Reply:
It's already added above: https://github.com/github/codeql/pull/17664/files#diff-65f5da78cea2691a3682e464d58f0ed3f31dec172341765c46b9b2ec6f6aa4deR222, but I'll add an explicit todo here too.